osu.Desktop/Overlays/VersionManager.cs

```diff
@@ -112,27 +113,35 @@ protected override void Dispose(bool isDisposing)
 
         private async void updateChecker()
         {
-            updateManager = await UpdateManager.GitHubUpdateManager(@"https://github.com/ppy/osu", @"osulazer", null, null, true);
+            try
+            {
+                updateManager = await UpdateManager.GitHubUpdateManager(@"https://github.com/ppy/osu", @"osulazer", null, null, true);
 
-            if (!updateManager.IsInstalledApp)
-                return;
+                if (!updateManager.IsInstalledApp)
+                    return;
 
-            var info = await updateManager.CheckForUpdate();
-            if (info.ReleasesToApply.Count > 0)
-            {
-                ProgressNotification n = new UpdateProgressNotification
+                var info = await updateManager.CheckForUpdate();
+                if (info.ReleasesToApply.Count > 0)
                 {
-                    Text = @"Downloading update..."
-                };
-                Schedule(() => notification.Post(n));
-                Schedule(() => n.State = ProgressNotificationState.Active);
-                await updateManager.DownloadReleases(info.ReleasesToApply, (int p) => Schedule(() => n.Progress = p / 100f));
-                Schedule(() => n.Text = @"Installing update...");
-                await updateManager.ApplyReleases(info, (int p) => Schedule(() => n.Progress = p / 100f));
-                Schedule(() => n.State = ProgressNotificationState.Completed);
+                    ProgressNotification n = new UpdateProgressNotification
+                    {
+                        Text = @"Downloading update..."
+                    };
+                    Schedule(() => notification.Post(n));
+                    Schedule(() => n.State = ProgressNotificationState.Active);
+                    await updateManager.DownloadReleases(info.ReleasesToApply, (int p) => Schedule(() => n.Progress = p / 100f));
+                    Schedule(() => n.Text = @"Installing update...");
+                    await updateManager.ApplyReleases(info, (int p) => Schedule(() => n.Progress = p / 100f));
+                    Schedule(() => n.State = ProgressNotificationState.Completed);
 
+                }
+                else
+                {
+                    //check again every 30 minutes.
+                    Scheduler.AddDelayed(updateChecker, 60000 * 30);
+                }
             }
-            else
+            catch(Exception e)

```

Comment:
can you change this to catch the specific exception related to not having network connectivity?